packages/animations/test/page_transition_switcher_test.dart

```diff
@@ -0,0 +1,622 @@
+    expect(_primaryAnimation[containerOne], moreOrLessEquals(0.4));
+    expect(_secondaryAnimation[containerOne], moreOrLessEquals(0.0));
+    expect(_primaryAnimation[containerTwo], moreOrLessEquals(1.0));
+    expect(_secondaryAnimation[containerTwo], moreOrLessEquals(0.8));
+    expect(_primaryAnimation[containerThree], moreOrLessEquals(0.2));
+    expect(_secondaryAnimation[containerThree], moreOrLessEquals(0.0));
+    await tester.pumpAndSettle();
+  });
+
+  testWidgets("doesn't transition in a new child of the same type.",
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: Container(color: const Color(0x00000000)),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    expect(find.byType(FadeTransition), findsOneWidget);
+    expect(find.byType(ScaleTransition), findsOneWidget);
+    FadeTransition fade = tester.firstWidget(find.byType(FadeTransition));
+    ScaleTransition scale = tester.firstWidget(find.byType(ScaleTransition));
+    expect(fade.opacity.value, equals(1.0));
+    expect(scale.scale.value, equals(1.0));
+
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: Container(color: const Color(0xff000000)),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(find.byType(FadeTransition), findsOneWidget);
+    expect(find.byType(ScaleTransition), findsOneWidget);
+    fade = tester.firstWidget(find.byType(FadeTransition));
+    scale = tester.firstWidget(find.byType(ScaleTransition));
+    expect(fade.opacity.value, equals(1.0));
+    expect(scale.scale.value, equals(1.0));
+    await tester.pumpAndSettle();
+  });
+
+  testWidgets('handles null children.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const PageTransitionSwitcher(
+        duration: Duration(milliseconds: 100),
+        child: null,
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    expect(find.byType(FadeTransition), findsNothing);
+    expect(find.byType(ScaleTransition), findsNothing);
+
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: Container(color: const Color(0xff000000)),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 40));
+    expect(find.byType(FadeTransition), findsOneWidget);
+    expect(find.byType(ScaleTransition), findsOneWidget);
+    FadeTransition fade = tester.firstWidget(find.byType(FadeTransition));
+    ScaleTransition scale = tester.firstWidget(find.byType(ScaleTransition));
+    expect(fade.opacity.value, moreOrLessEquals(1.0));
+    expect(scale.scale.value, moreOrLessEquals(0.4));
+    await tester.pumpAndSettle();
```

Comment:
Maybe add a comment here saying that this `pumpAndSettle` call is meant to let the entering widget's transition continue to completion